src/interface/motor.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+namespace interface {
+class Motor {
+ public:
+  virtual void Move(int) = 0;
+  virtual void MoveVoltage(int) = 0;
+  virtual void MoveVelocity(float) = 0;
+  virtual void MoveAbsolute(double position, int max_velocity) = 0;
```

Comment:
MoveVelocity and MoveAbsolute are two interesting functions, as they rely on closed-loop control and likely will require a motor to also have an encoder present to perform properly. 

I'm interested in your thoughts here - do you think these are fitting to have inside of the general motor interface? My thought is that if we try to implement a non-vex motor (ex. a 775), we won't have feedback unless the motor also adds a controller in the loop. I think we're making an implicit dependency within the motor interface on the motor also being an encoder (or having an encoder accessible), which is something that the pros motors do to their interfaces internally.

I'm definitely alright with leaving this, but I think if our plan is to expand this library outside of vex, this may cause some difficulty in the future

Reply:
Yeah I haven't been thrilled by Motor having 4 methods, primarily since many implementations will only use one or two. I left it as four primarily since I haven't dove into what is actually useful yet.

A motor with feedback is a bit of an interesting case, however, since the current interfaces don't really support it - a ProsMotor which uses its encoder requires the Motor and Encoder to be decoupled in implementations. So it is challenging to create a motor which recieves feedback from an encoder; you'd have to hook the two together manually in the implementation. Maybe it's worthwhile to create a specific MotorAndEncoder interface which adds closed loop controls? That's a case where you would want to have a sensor working directly with a motor.